web3/_utils/miner.py

```diff
@@ -18,49 +19,62 @@
     Wei,
 )
 
-make_dag: Method[Callable[[BlockNumber], bool]] = Method(
+#
+# The Geth client deprecated the miner namespace.
+#
+
+_make_dag: Method[Callable[[BlockNumber], bool]] = Method(
     RPC.miner_makeDag,
     mungers=[default_root_munger],
 )
 
+make_dag = DeprecatedMethod(_make_dag, "make_dag", "deprecated")
```

Comment:
Nice use of `DeprecatedMethod`! I think since the only place we're using it is in this file, we can change it to better suit our purposes. Because we're not replacing the method with anything, we can update the warning [here](https://github.com/ethereum/web3.py/blob/main/web3/method.py#L250) to say something along the lines of f"{self.old_name} is no longer supported." 

Or, if you really wanted to go for some extra credit, you could change the `DeprecatedMethod` class to take a new, optional argument called `message` and then on `__get__` we could return the message if there is one. So the new DeprecatedMethod class might look something like:

```

class DeprecatedMethod:
    def __init__(
        self,
        method: Method[Callable[..., Any]],
        old_name: str,
        new_name: Optional[str] = None,
        msg: Optional[str] = None,
    ) -> None:
        self.method = method
        self.old_name = old_name
        self.new_name = new_name
        self.msg = msg

    def __get__(
        self, obj: Optional["Module"] = None, obj_type: Optional[Type["Module"]] = None
    ) -> Any:
        if self.msg is not None:
            warnings.warn(self.msg)
        else:
            warnings.warn(
                f"{self.old_name} is deprecated in favor of {self.new_name}",
                category=DeprecationWarning,
            )
        return self.method.__get__(obj, obj_type)
```

And this line would look something like:
```suggestion
make_dag = DeprecatedMethod(_make_dag, "make_dag", msg="All mining methods have been deprecated")
```